### PR TITLE
Cmdlet documentation using PlatyPS

### PR DESCRIPTION
**Pull request by davegreen**
An initial attempt at #133 to introduce PlatyPS documentation generation into the psake build script.

This pull request also adds PlatyPS documentation options to the example module template and an initial about topic for creating a plaster manifest.

### CODE REVIEW

File: examples/NewModuleTemplate/build.psake.ps1 [hunk 1]
```diff
@@ -273,6 +277,27 @@ Task Test -depends Build {
...
+        $DefaultLocale = 'en-US'
+    }
+
+    Import-Module "$PublishDir\$ModuleName.psd1" -Global
+    
+    (Get-ChildItem -Path $DocsRootDir -Directory).FullName | ForEach-Object {
+        Update-MarkdownHelp -Path $_ | Out-Null
+    }
+
+    New-MarkdownHelp -Module $ModuleName -Locale $DefaultLocale -OutputFolder "$DocsRootDir\$DefaultLocale" -WithModulePage -ErrorAction SilentlyContinue | Out-Null
```
Comment (rkeithhill) [inline examples/NewModuleTemplate/build.psake.ps1]: The double quotes on the OutputFolder arg are not necessary. You can remove those.

File: examples/NewModuleTemplate/plasterManifest.xml [hunk 2]
```diff
@@ -75,6 +78,9 @@ Scaffold a PowerShell Module with the files required to run Pester tests, build
         <file source='build*.ps1'
               destination=''
               condition='$PLASTER_PARAM_Options -contains "PSake"'/>
+        <file source=''
+              destination='docs\'
+              condition='$PLASTER_PARAM_Options -contains "PlatyPS"'/>
```
Comment (rkeithhill) [inline examples/NewModuleTemplate/plasterManifest.xml]: Ditto on casing.

File: examples/NewModuleTemplate/plasterManifest.xml [hunk 3]
```diff
@@ -156,6 +162,9 @@ Scaffold a PowerShell Module with the files required to run Pester tests, build
         <requireModule name="psake" condition='$PLASTER_PARAM_Options -contains "PSake"'
             message="Without psake, you will not be able to run the provided build script to build and/or publish your module."/>
 
+        <requireModule name="PlatyPS" condition='$PLASTER_PARAM_Options -contains "PlatyPS"'
```
Comment (rkeithhill) [inline examples/NewModuleTemplate/plasterManifest.xml]: Ditto on casing for this line and the next.

File: examples/NewModuleTemplate/plasterManifest.xml [hunk 4]
```diff
@@ -156,6 +162,9 @@ Scaffold a PowerShell Module with the files required to run Pester tests, build
...
+            message="Without PlatyPS, you will not be able to generate PowerShell external help for your module using markdown."/>
+
```
Comment (rkeithhill) [inline examples/NewModuleTemplate/plasterManifest.xml]: Ditto on casing.

File: examples/NewModuleTemplate/plasterManifest.xml [hunk 5]
```diff
@@ -171,5 +180,12 @@ A Pester test has been created to validate the module's manifest file.  Add addi
 You can run the Pester tests in your project by executing the 'test' task by pressing Ctrl+P, then type 'task test'.
 
         </message>
+
+        <message condition="$PLASTER_PARAM_Options -contains 'PlatyPS'">
```
Comment (rkeithhill) [inline examples/NewModuleTemplate/plasterManifest.xml]: Ditto on casing.

File: docs/en-US/about_Plaster_CreatingAManifest.help.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
+# Creating a Plaster Manifest
```
Comment (rkeithhill) [inline docs/en-US/about_Plaster_CreatingAManifest.help.md]: Awesome!! Thanks for putting this together.  This is a great start.

File: build.psake.ps1 [hunk 7]
```diff
@@ -273,6 +277,27 @@ Task Test -depends Build {
...
+        $DefaultLocale = 'en-US'
+    }
+
+    Import-Module "$PublishDir\$ModuleName.psd1" -Global
+    
+    (Get-ChildItem -Path $DocsRootDir -Directory).FullName | ForEach-Object {
+        Update-MarkdownHelp -Path $_ | Out-Null
+    }
+
+    New-MarkdownHelp -Module $ModuleName -Locale $DefaultLocale -OutputFolder "$DocsRootDir\$DefaultLocale" -WithModulePage -ErrorAction SilentlyContinue | Out-Null
```
Comment (rkeithhill) [inline build.psake.ps1]: The double quotes on the Path arg are unnecessary. You can remove those.

File: build.psake.ps1 [hunk 8]
```diff
@@ -273,6 +277,27 @@ Task Test -depends Build {
...
+    Remove-Module $ModuleName
+}
+
+Task BuildDocs -depends GenerateDocs {
+    foreach ($locale in (Get-ChildItem -Path $DocsRootDir -Directory).Name) {
+        New-ExternalHelp -Path "$DocsRootDir\$locale" -OutputPath "$PublishDir\$locale" -Force -ErrorAction SilentlyContinue | Out-Null
```
Comment (rkeithhill) [inline build.psake.ps1]: The double quotes on the Path arg are unnecessary. You can remove those.

**Comment by daviwil** [comment]
Dave, thanks so much for getting this work started, it's a huge help!  I'm in agreement with Keith's comments, but overall looks good!

**Comment by davegreen** [comment]
Good stuff. Glad to help out.

**Pull request merged**